packages/app/public/service-worker.js

```diff
@@ -0,0 +1,79 @@
+// urlB64ToUint8Array is a magic function that will encode the base64 public key
+// to Array buffer which is needed by the subscription option
+const urlB64ToUint8Array = (base64String) => {
+  const padding = "=".repeat((4 - (base64String.length % 4)) % 4);
+  const base64 = (base64String + padding)
+    .replace(/\-/g, "+")
+    .replace(/_/g, "/");
+  const rawData = atob(base64);
+  const outputArray = new Uint8Array(rawData.length);
+  for (let i = 0; i < rawData.length; ++i) {
+    outputArray[i] = rawData.charCodeAt(i);
+  }
+  return outputArray;
+};
+
+// todo: figure out how to do this
+const ORIGIN = `http://localhost:3000`;
```

Comment:
is there a way to do this?

Reply:
@dajinchu do you know how to get the origin on the frontend? for some reason, doing `window.origin` is blocked in the service-worker thread.

Reply:
is it possible to just send requests to '/whatever' or is the fully qualified path required?